### PR TITLE
feat(ledger-vault): injectable DMK transport and Speculos presign cov…

### PR DESCRIPTION
**Pull request by jrwbabylonlab**
Adds a transport-factory seam to dmkSession (web-hid stays the default) so the DMK path can be driven against Speculos, plus two package e2e stages: depositor-graph presign (Payout + per-challenger NoPayout, fixtures built by the production SDK builders) and abort/dispatcher recovery, which firmware-pins the behaviour T3's cancel recovery assumes. Run live against fix/audit-findings @ 29beb88d5 — existing stages 24/24.
Two findings came out of the live run and are not fixed here: a firmware bug where _validate_nopayout routes the vault group by PegIn txid while protocol NoPayout spends Assert:0 (raised with Ledger; stage 11 pins both shapes), and a host gap where the expected-signature table demands a payout input-1 yield the device never produces under a loaded intent.

Also fixes a latent test-fixture bug: DEPOSITOR_PATH was built with a bitwise OR, yielding a negative int32 that production's assertBip86Path rejects — wire-identical, so live-run-only impact, now pinned by an offline unit test.

### CODE REVIEW

File: packages/babylon-ledger-vault-signer/package.json [hunk 1]
```diff
@@ -44,7 +44,10 @@
     "rxjs": "7.8.2"
   },
   "devDependencies": {
+    "@babylonlabs-io/babylon-tbv-rust-wasm": "workspace:*",
+    "@babylonlabs-io/ts-sdk": "workspace:*",
```
[the remark this answers is not in the document]
Comment (jrwbabylonlab) [reply packages/babylon-ledger-vault-signer/package.json:48]: not planning — `workspace:*` is the monorepo convention for internal packages; the pin rule targets external `^` ranges

File: packages/babylon-ledger-vault-signer/src/dmkSession.ts [hunk 2]
```diff
@@ -29,37 +37,111 @@ export interface DmkSessionHandle {
...
+ *
+ * @throws while a DMK singleton exists (built or building): each build captures
+ * the factory it was given, so a late swap would leave discovery pointed at a
+ * transport that DMK does not hold. Call {@link closeDmk} first.
+ */
+export function setDmkTransportOverride(override: DmkTransportOverride | undefined): void {
+  if (dmkPromise !== undefined) {
+    throw new Error("setDmkTransportOverride: DMK already built or building — closeDmk() before changing transports");
+  }
+  transportOverride = override;
```
[the remark this answers is not in the document]
Comment (jrwbabylonlab) [reply packages/babylon-ledger-vault-signer/src/dmkSession.ts:84]: not planning — inline messages match this package's style; the env keys are each read once

**Review by jonybur — APPROVED** [review]
The DMK transport seam is well-designed. The publish() invariant catches the closeDmk-mid-build race and refuses to adopt an orphan whose successor already owns the memo; the never-closes-orphaned-build test states the reasoning positively — build() only stores the factory, so close() would construct the transport and leak listeners. The DEPOSITOR_PATH bitwise-OR fix is exactly the kind of latent test-fixture bug worth the ungated unit that pins it. Stage 11 (depositor-graph presign) exercises the post-#2281 narrowing on real firmware and cross-checks with the SDK's own script-path verifier; stage 12 (abort/dispatcher recovery) is impressive coverage — pinning the eaten-APDU 0x6a80, the pre_pegin_signed++ cap-burn, and the full re-ceremony recovery. The two firmware-divergence pins (NoPayout routing, payout input-1 yield) are correctly documented as divergences to raise upstream, not host bugs to hide. fileParallelism gate prevents cross-suite Speculos collisions.
